### PR TITLE
Tabs Component

### PR DESCRIPTION
**Pull request by rtkaaho**
- Tabs component for custom view management via state or react-router

Screen Shot 2020-03-24 at 11 55 09 AM

### CODE REVIEW

File: src/components/tabs/TabsContext.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export interface TabsContextProps {
+  setActiveItem?: any; // TODO add type
```
Comment (rtkaaho) [inline src/components/tabs/TabsContext.tsx]: Neeed type here

File: src/components/tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        align-items: center;
+      `}
+  `}
+`;
+
+const Inkbar = styled.div<{ theme: GlobalTheme }>`
+  ${({ theme }) => css`
+    position: relative;
+    background: ${theme.colors.primary};
+    height: 4px;
```
Comment (rtkaaho) [inline src/components/tabs/Tabs.tsx]: add to theme

File: src/components/tabs/TabsItem.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  /** title of the tab item */
+  title: React.ReactNode;
+}
+
+const Container = styled.div``;
+
+const Title = styled(Typography.Body)<{ theme: any; isSelected: boolean }>`
+  ${({ theme, isSelected }) => css`
+    user-select: none;
+    padding: 8px 16px;
```
Comment (rtkaaho) [inline src/components/tabs/TabsItem.tsx]: add to theme

File: src/components/tabs/__tests__/Tabs.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { Tabs } from '../Tabs';
+
+describe('Tabs', () => {
+  it('renders', () => {
+    // your test case here
```
Comment (rtkaaho) [inline src/components/tabs/__tests__/Tabs.spec.tsx]: add test cases

File: src/components/tabs/TabsContext.tsx (hunk 1 above)
Comment (rtkaaho) [reply src/components/tabs/TabsContext.tsx]: done!

File: src/components/tabs/Tabs.tsx (hunk 2 above)
Comment (rtkaaho) [reply src/components/tabs/Tabs.tsx]: done!

File: src/components/tabs/TabsItem.tsx (hunk 3 above)
Comment (rtkaaho) [reply src/components/tabs/TabsItem.tsx]: done!

File: src/components/tabs/__tests__/Tabs.spec.tsx (hunk 4 above)
Comment (rtkaaho) [reply src/components/tabs/__tests__/Tabs.spec.tsx]: done!

File: src/components/tabs/Tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      <Container className={`${className} rtk-tabs`}>
+        <TabsContainer
+          className="rtk-tabs-container"
+          placement={placement}
+          theme={theme}
+        >
+          {children}
+        </TabsContainer>
+        {activeItem == null ? null : (
+          <Inkbar
```
Comment (rtktpeng) [inline src/components/tabs/Tabs.tsx]: I find this highlighting approach of active items very refreshing. 😄

**Comment by rtkaaho** [comment]
Ouu fair point, I will suppress those lint warnings in the next pr :) thanks!

File: src/components/tabs/Tabs.tsx (hunk 5 above)
Comment (rtkaaho) [reply src/components/tabs/Tabs.tsx]: :)

**Pull request merged**